reactor-extra/src/main/java/reactor/ui/JavaFxScheduler.java

```diff
@@ -0,0 +1,262 @@
+	}
+
+	@Override
+	public Disposable schedulePeriodically(Runnable task, long initialDelay, long period, TimeUnit unit) {
+		JavaFxWorker javaFxWorker = new JavaFxWorker();
+		long initialDelayMillis = unit.toMillis(initialDelay);
+
+		SwtPeriodicDirectAction a = new SwtPeriodicDirectAction(task, javaFxWorker,
+				System.currentTimeMillis() + initialDelayMillis, unit.toMillis(period));
+
+		if (initialDelay <= 0) {
+			javaFxWorker.schedule(a);
+		} else {
+			javaFxWorker.schedule(a, initialDelayMillis, TimeUnit.MILLISECONDS);
+		}
+
+		return a;
+	}
+
+	@Override
+	public Worker createWorker() {
+		return new JavaFxWorker();
+	}
+
+	/**
+	 * A Worker implementation which manages a queue of QueuedRunnable for execution
+	 * on the Java FX Application thread For a simpler implementation the queue
+	 * always contains at least one element. {@link #head} is the element, which is
+	 * in execution or was last executed {@link #tail} is an atomic reference to the
+	 * last element in the queue, or null when the worker was disposed Recursive
+	 * actions are not preferred and inserted at the tail of the queue as any other
+	 * action would be The Worker will only schedule a single job with
+	 * {@link Platform#runLater(Runnable)} for when the queue was previously empty
+	 */
+	private static class JavaFxWorker implements Worker, Runnable {
+		private volatile QueuedRunnable head = new QueuedRunnable(null); /// only advanced in run(), initialised with a
+																			/// starter element
+		private final AtomicReference<QueuedRunnable> tail = new AtomicReference<>(head); /// points to the last
+																							/// element, null when
+																							/// disposed
+
+		private static class QueuedRunnable extends AtomicReference<QueuedRunnable> implements Disposable, Runnable {
+			private volatile Runnable action;
+
+			private QueuedRunnable(Runnable action) {
+				this.action = action;
+			}
+
+			@Override
+			public void dispose() {
+				action = null;
+			}
+
+			@Override
+			public boolean isDisposed() {
+				return action == null;
+			}
+
+			@Override
+			public void run() {
+				Runnable action = this.action;
+				if (action != null) {
+					action.run();
+				}
+				this.action = null;
+			}
+		}
+
+		@Override
+		public void dispose() {
+			tail.set(null);
+			QueuedRunnable qr = this.head;
+			while (qr != null) {
+				qr.dispose();
+				qr = qr.getAndSet(null);
+			}
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return tail.get() == null;
+		}
+
+		@Override
+		public Disposable schedule(final Runnable action, long delayTime, TimeUnit unit) {
+			long delay = Math.max(0, unit.toMillis(delayTime));
+			assertThatTheDelayIsValidForTheJavaFxTimer(delay);
+
+			final QueuedRunnable queuedRunnable = new QueuedRunnable(action);
+			if (delay == 0) { // delay is too small for the java fx timer, schedule it without delay
+				return schedule(queuedRunnable);
+			}
+
+			final Timeline timer = new Timeline(
```

Comment:
Note that the timing of executing the keyframe is only loosely guaranteed. But I guess for the context of a JavaFX application this is not really different from using the Animation/Timeline directly.